sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTypes.scala

```diff
@@ -289,7 +295,20 @@ private[parquet] object ParquetTypesConverter extends Logging {
       name: String,
       nullable: Boolean = true,
       inArray: Boolean = false,
+      parquetSchema: Option[ParquetType] = None,
       toThriftSchemaNames: Boolean = false): ParquetType = {
+
+    val parquetElementTypeBySchema =
```

Comment:
It also performs a type check / conversion. That's why I've removed it. It would look like this

```
val parquetElementTypeBySchema =
  parquetSchema.filter(_.isInstanceOf[ParquetGroupType]).filter(_.containsField(name)).map(_.getType(name))
```

I would settle on collect, does that look ok?

```
  val parquetElementTypeBySchema = parquetSchema.collect {
    case gType : ParquetGroupType if (gType.containsField(name)) => gType.getType(name)
  }
```